cli/test/integration/p2p/connect_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package p2p
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/sourcenetwork/defradb/cli/test/action"
+	"github.com/sourcenetwork/defradb/cli/test/integration"
+)
+
+var (
+	peerIDs = []string{
+		"12D3KooWGj2wWqxSKB2Lqg6K6ye7N7W4YcpoeNpdXakHkgGjUqHC",
+		"12D3KooWBVR4AGs7pTwdjUH1Pd9bgh85hyf1xtV3rg7bj3VfZGoT",
+	}
+	invalidPeerID = "invalid-peer-id"
+	addresses     = []string{
+		fmt.Sprintf("/ip4/127.0.0.1/tcp/9000/p2p/%s", peerIDs[0]),
+		fmt.Sprintf("/ip4/127.0.0.1/tcp/9001/p2p/%s", peerIDs[1]),
+	}
+	addressWithInvalidPeerID = fmt.Sprintf("/ip4/127.0.0.1/tcp/9000/p2p/%s", invalidPeerID)
+	addressWithInvalidIP     = "/ip4/999.999.999.999/tcp/9000/p2p/12D3KooWGj2wWqxSKB2Lqg6K6ye7N7W4YcpoeNpdXakHkgGjUqHC"
```

Comment:
thought: I dislike having these here, away from the test declarations, but I assume most will move once support for multiple nodes in the cli tests is added.

question: What actually broke when you added multiple?  I thought it would work and was already used (it is not, I must have mixed memories with Lens).

Reply:
> thought: I dislike having these here, away from the test declarations, but I assume most will move once support for multiple nodes in the cli tests is added.

The exact address is not important to the tests other than the fact that its valid or not and they are reused across all the p2p tests. So it made sense to me to have them defined this way.

> question: What actually broke when you added multiple? I thought it would work and was already used (it is not, I must have mixed memories with Lens).

Can you clarify this question please. Not sure what you are referring to.

Reply:
> Can you clarify this question please. Not sure what you are referring to.

There are a couple of tests with the below comment, I read it as you tried to setup multiple nodes and the test action couldn't handle it somehow.

> // TODO: Add capability to have multiple defradb instances in tests, so we can
// actually test successful connections. #4021